### DISCUSSION TITLE
Add "I'm feeling lucky" functionality to Results webpart

### DISCUSSION THREAD

**Discussion by SjoerdV** [microsoft-search/pnp-modern-search#1973] [Ideas]
**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
It would be really nice if  "I'm feeling lucky" functionality could be implemented. It would work like this: When a single search result is found for any given query this result is expanded/opened immediately (no delay), on multiple results nothing happens. In effect this would work like a 'dynamic redirect' where the target location differs on the item found in the Results webpart. 
Alternatively it would open the first result of any query also the ones resulting in multiple hits (maybe setting a delay is needed in this case).
In any case this would prevent the need to click any result for the user in such 'lucky' cases. Less clicking = happier users

**Describe alternatives you've considered**
Another custom spfx app/webpart maybe, but I'd really like it integrated in the existing search solution as an option in the results webpart.

**Additional context**
Should be clear I hope

**Comment by FranckyC** [comment]
Cool idea but I'm not really sure about the user experience. Something that can be more misleading for users in the end. Also cost vs value doesn't seem to be worth it to gain only 1 click :/.

**Reply by SjoerdV** [reply]
Well, OK I understand. But don't underestimate the impact this kind of 'smart redirect' logic might have, as we currently do not even have the same functionality in Modern pages as we did in Classic pages where a simple javascript in a CEWP combined with an URL parameter was very useful. This would actually go one step further as it would return all properties of a resulted item to construct the target redirect URL with,very powerful! 

You could even use this concept to link certain unrelated entities inside a site (like a List with items to documents inside a Doclib without using 'still a bit awkward' Lookup columns but just by using a different common field value.

Using ```?Mode=Edit``` on the page containing the redirect configuration would of course not trigger the redirect, so should not really be an issue concerning usability imho. 

I would only actually use it on a page dedicated to redirection, so not inside a full Search page with verticals, filters, results, searchbox etc.

**Reply by wobba** [reply]
I think a custom web component would be a good solution. With this you could check the number of results and redirect as you wish based on the logic you define.

**Reply by SjoerdV** [reply]
Ok, so you would recommend an external solution for this, if I understand you correctly?

**Reply by wobba** [reply]
Correct.

**Reply by SjoerdV** [reply]
Right on, now commencing the search for a developer, who would appreciate getting into this kind of feature :-)

**Reply by FranckyC** [reply]
I do some consultancy if you really want this feature ;)

**Reply by SjoerdV** [reply]
I was hoping more a long the lines of: I desperately need a hobby project :-)
